impl/ocaml/mariadb/sqlgg_mariadb.ml

```diff
@@ -288,7 +289,8 @@ let execute db sql set_params =
   with_stmt db sql @@ fun stmt ->
   let open IO in
   set_params stmt >>=
-  fun res -> return (Int64.of_int (M.Res.affected_rows res))
+  fun res -> return { affected_rows = Int64.of_int (M.Res.affected_rows res); insert_id = Int64.of_int (M.Res.insert_id res) }
```

Comment:
why do Int64.of_int? it is less convenient type for no benefit

Reply:
This is just me blindly copying the current code 🙈. I switched `insert_id` to `int` in [`12ea564` (#140)](https://github.com/ygrek/sqlgg/pull/140/commits/12ea5649f79d4fc5445a790fd7b150a8f02ea588)

I got curious about `affected_rows`, it was added 15 years ago in https://github.com/ygrek/sqlgg/commit/c9e8d3f33380ea1ad8a92e72c9b8ae9ebaee4d3c#diff-6d600b7526702e1d382e1022fdf50144080017d3da236b6c4dc96b33529bb19b. At that time (and before this PR), there was no way to use a different `execute_response` type for each implementation and [`mysql` return `int64`](https://github.com/ygrek/ocaml-mysql/blob/master/mysql.mli#L247) so I believe that's the reason.

Since we can now specify the `execute_response` by implementation, I changed it to be `int` for `mariadb` and `sqlite`. It's a breaking change, but since there was one already I guess that's fine

Reply:
And now I'm realizing that `mysql` has also binding for [insert_id](https://github.com/ygrek/ocaml-mysql/blob/master/mysql.mli#L252). I adapted the `execute_response` type to also include `insert_id` [`6793f79` (#140)](https://github.com/ygrek/sqlgg/pull/140/commits/6793f792aa72f58c36dd60c38df68b3fd709c7f7)

Reply:
I'm confused about the difference between `mysql` and `mariadb`; both bind to `mysql_stmt_affected_rows` 🤔 Looks like there's a difference between mysql version? https://dev.mysql.com/doc/c-api/8.4/en/mysql-stmt-affected-rows.html vs https://dev.mysql.com/doc/c-api/5.7/en/mysql-stmt-affected-rows.html

Reply:
Ok looks like they're probably the same after all

> * my_ulonglong: This is a MySQL-specific typedef, which has been used traditionally in older versions (like 5.7) to represent 64-bit unsigned integers.
> * uint64_t: This is a standard C typedef introduced by the C99 standard, representing a 64-bit unsigned integer and is more widely recognized outside of MySQL.

But why `mysql` and `mariadb` use a different `OCaml` type. Which one is correct? Is it related to https://github.com/ocaml-community/ocaml-mariadb/issues/31 ?

Reply:
> But why mysql and mariadb use a different OCaml type. Which one is correct? 

it is different libraries, they made different choices at some point, correct depends on how you define correct :)